src/enqueue.cc

```diff
@@ -2288,12 +2288,19 @@ ncclResult_t ncclRedOpCreatePreMulSum_impl(ncclRedOp_t *op, void *scalar, ncclDa
   }
   *op = ncclRedOp_t(int(ncclNumOps) + ix);
   *op = ncclUserRedOpMangle(comm, *op);
+
+  // ! recording at sink
+  rcclApiCall call({.opName = "RedOpCreatePreMulSum", .sendbuff = scalar, .datatype = datatype,
+		    .op = *op, .root = residence, .comm = comm}, rrRedOpCreatePreMulSum);
+  NCCLCHECK(Replayer::instance().record(call));
   TRACE_CALL("ncclRedOpCreatePreMulSum(%d,%p,%d,%d,%p)", *op, scalar, datatype, residence, comm);
   return ncclSuccess;
 }
 
 NCCL_API(ncclResult_t, ncclRedOpDestroy, ncclRedOp_t op, ncclComm_t comm);
 ncclResult_t ncclRedOpDestroy_impl(ncclRedOp_t op, ncclComm_t comm) {
+  rcclApiCall call({.opName = "RedOpDestroy", .op = op, .comm = comm}, rrRedOpDestroy);
```

Comment:
Having both opName and rrRedOpDestroy is redundant.
Having a table that allows for conversion between the enumeration and string name would be better.